tools/pylib/_boutpp_build/backend.py

```diff
@@ -0,0 +1,216 @@
+import os  # corelib
+import glob  # corelib
+import hashlib  # corelib
+import base64  # corelib
+import tempfile  # corelib
+import subprocess  # corelib
+
+try:
+    import packaging.tags  # packaging
+except:
+    packaging = None
+
+
+def run(cmd):
+    print(f"running `{cmd}`")
+    ret = os.system(cmd)
+    assert ret == 0, f"{cmd} failed with {ret}"
+
+
+useLocalVersion = True
+
+
+def getversion(_cache={}):
+    if "r" not in _cache:
+        try:
+            tmp = run2("git describe --tags --match=v4.0.0")
+            if useLocalVersion:
+                tmp = run2(
+                    f"echo {tmp.strip()} | sed -e s/v4.0.0-/v5.0.0.dev/ -e s/-/+/"
+                )
+            else:
+                tmp = run2(
+                    f"echo {tmp.strip()} | sed -e s/v4.0.0-/v5.0.0.dev/ -e s/-.*//"
+                )
+            _cache["r"] = tmp
+            with open("_version.txt", "w") as f:
+                f.write(_cache["r"])
+        except AssertionError:
+            _cache["r"] = run2("cat _version.txt")
+    return _cache["r"].strip()
+
+
+def run2(cmd):
+    child = subprocess.Popen(
+        cmd, stderr=subprocess.STDOUT, stdout=subprocess.PIPE, shell=True
+    )
+    output = child.stdout.read().decode("utf-8", "ignore")
+    child.communicate()
+    assert child.returncode == 0, f"{cmd} failed with {child.returncode}"
```

Comment:
[`subprocess.run`](https://docs.python.org/3/library/subprocess.html#subprocess.run) has a much simpler interface:

```suggestion
    output = subprocess.run(cmd, capture_output=True, shell=True, check=True).stdout
```

Reply:
Great :tada: